R-package/configure

```diff
@@ -1,6 +1,6 @@
 #! /bin/sh
 # Guess values for system-dependent variables and create Makefiles.
-# Generated by GNU Autoconf 2.69 for lightgbm 3.0.0.99.
+# Generated by GNU Autoconf 2.69 for lightgbm 3.1.0.
```

Comment:
/gha run r-valgrind

Reply:
this passed: https://github.com/microsoft/LightGBM/actions/runs/350733081

Reply:
/gha build r-artifacts

Reply:
looks like this job is broken:

> Error: Input 'submodules' not supported when falling back to download using the GitHub REST API. To create a local Git repository instead, add Git 2.18 or higher to the PATH.

run: https://github.com/microsoft/LightGBM/runs/1404264739?check_suite_focus=true

I opened https://github.com/microsoft/LightGBM/issues/3567. It doesn't have to be fixed for the release. I can build the package locally.